### PR TITLE
Add a sensible default for theme app extension template

### PR DESCRIPTION
**Pull request by Arkham**
### WHY are these changes introduced?

When you run `shopify generate extension --type=theme_app_extension` we generate an extension which is basically empty AND not deployable (the user has to manually remove all `.gitkeep` files).

Also fixes https://github.com/Shopify/internal-cli-foundations/issues/493

### WHAT is this pull request doing?

- Provide a sensible template for theme app extensions with:
  - 1 block
  - 1 snippet
  - 1 locale
  - 1 asset
- it also uses metafields for fun
- now the generated theme app extension can be deployed 

### How to test your changes?

- `bin/create-test-app.js -e theme`
- press `p` and install the app in your dev store
- wait for theme app extension to show you three links
- click on the first one and click "Enable"
- click on the theme editor and add a new section with your theme app extension
- open http://localhost:9292 to see the preview
- if you make a change in the source you should see it reflected

### Post-release steps

- change https://shopify.dev/docs/apps/online-store/theme-app-extensions/getting-started

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by alvaro-shopify** [comment]
could it make sense to include a README about how to use the template?
* Create the concrete metafield
* Configure the products
* .....

File: packages/app/templates/theme-extension/blocks/star_rating.liquid.raw [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{% schema %}
+{
+  "name": "Star Rating",
+  "target": "section",
+  "settings": [
+    {"type": "product", "id": "product", "label": "product", "autofill": true},
+    {"type": "color", "id": "colour", "label": "Star Colour", "default": "#ff0000"}
+  ]
+}
+{% endschema %}
```
Comment (karreiro) [inline packages/app/templates/theme-extension/blocks/star_rating.liquid.raw:18]: I've applied the Prettier Plugin in this file to make it even prettier:

```suggestion
{% assign avg_rating = block.settings.product.metafields.demo.avg_rating.value | round %}
<span style="color:{{ block.settings.colour }}">
  {% render 'stars', rating: avg_rating %}
</span>
{% if avg_rating >= 4 %}
  <br>
  <img src="{{ "thumbs-up.png" | asset_img_url: '15x' }}" height="15" width="15" loading="lazy">
  {{ 'ratings.home.recommendationText' | t }}
{% endif %}

{% schema %}
{
  "name": "Star Rating",
  "target": "section",
  "settings": [
    { "type": "product", "id": "product", "label": "product", "autofill": true },
    { "type": "color", "id": "colour", "label": "Star Colour", "default": "#ff0000" }
  ]
}
{% endschema %}

```

File: packages/app/templates/theme-extension/snippets/stars.liquid.raw [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{ "ratings.stars.label" | t }}:
+
+{%- for i in (1..rating) -%}
+★
+{%- endfor -%}
+{%- assign blank_stars = 5 | minus: rating -%}
+{%- for i in (1..blank_stars) -%}
+☆
+{%- endfor -%}
```
Comment (karreiro) [inline packages/app/templates/theme-extension/snippets/stars.liquid.raw:9]: I've applied the Liquid Prettier Plugin here as well:

```suggestion
{{ 'ratings.stars.label' | t }}:

{%- for i in (1..rating) -%}
  ★
{%- endfor -%}
{%- assign blank_stars = 5 | minus: rating -%}
{%- for i in (1..blank_stars) -%}
  ☆
{%- endfor -%}

```

**Review by karreiro — APPROVED** [review]
Thank you, @Arkham! Great stuff 🚀 

I've left only two minor comments applying the [Liquid Prettier Plugin](https://shopify.dev/docs/themes/tools/liquid-prettier-plugin) in some files.

**Pull request merged**